modules/oe_whitelabel_starter_event/src/Plugin/Field/FieldFormatter/EventDateRangeFormatter.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_whitelabel_starter_event\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\datetime\Plugin\Field\FieldFormatter\DateTimeFormatterBase;
+use Drupal\datetime_range\DateTimeRangeTrait;
+
+/**
+ * Plugin implementation of the 'Event dates' formatter for 'daterange' fields.
+ *
+ * This formatter renders the data range as plain text, with a fully
+ * configurable date format using the PHP date syntax and separator.
+ *
+ * @FieldFormatter(
+ *   id = "event_date_range_format",
+ *   label = @Translation("Event date range format"),
+ *   field_types = {
+ *     "daterange"
+ *   }
+ * )
+ */
+class EventDateRangeFormatter extends DateTimeFormatterBase {
```

Comment:
I forgot to mention: I don't think this formatter is specific to events.
We should change the plugin id and the class name, and also move it into oe_whitelabel_helper. Possibly even oe_bootstrap_theme_helper - TBD.
Basically the formatter does this:
- If same day, it shows `$start_date->format($day_format) . $separator0 . $start_time->format($time_format) . $time_separator . $end_time->format($time_format)`.
- If different day, it shows `$start_date->format($datetime_format) . $day_separator . $end_date->format($datetime_format)`.

The formats and the separators should be configurable and translatable.

This sounds quite generic, and is similar to what e.g. https://www.drupal.org/project/daterange_compact promises to do. (I am not saying we should use that contrib module, I just want to show this use case is quite generic.)